platform/tests/config/config_test.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+
+addr="-s command-line"
+
+# 1 - pwd, home, cli
+test_pwd_home_cli() {
+  # for config file in current working directory
+  mkdir -p $PWD/.amp
+  echo "Server: present" > $PWD/.amp/amp.yml
+
+  # for config file in home directory
+  mkdir -p $HOME/.config/amp
+  echo "Server: home" > $HOME/.config/amp/amp.yml
+
+  # server address passed as command line argument
+  amp $addr version | grep -Eq "Server:[[:space:]]+command-line"
+
+  rm -R $PWD/.amp
+  rm -R $HOME/.config/amp
```

Comment:
This won't work. The test will pass as long as the last command passes. If you want to clean up at the end of your test, just make sure to preserve the result of the previous grep (`local ec=$?`) and then return it at the end of the function (`return $ec`).